piquasso/_backends/gaussian/state.py

```diff
@@ -1073,60 +1068,29 @@ def _get_particle_number_choice_and_probability(
 
         d = len(self)
 
-        Q = (self.complex_covariance + np.identity(2 * d)) / 2
-
-        Qinv = np.linalg.inv(Q)
-
-        B = - Qinv[d:, :d]
+        B = - np.linalg.inv(self.Q_matrix)[d:, :d]
```

Comment:
This refactor is very efficient (eliminating all the calculations with A matrix) but is not so clear from the code. Does it appear in the documentation?

Reply:
Yeah, it is in the docstring!